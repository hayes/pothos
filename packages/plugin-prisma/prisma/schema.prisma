// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema
datasource db {
  provider = "sqlite"
  url      = "file:./dev.db"
}

generator client {
  provider = "prisma-client-js"
  output   = "../tests/client"
}

generator pothos {
  provider          = "node -r @swc-node/register ./src/generator.ts"
  clientOutput      = "../tests/client"
  pluginPath        = "../src"
  output            = "../tests/generated.ts"
  generateDatamodel = true
  documentation     = true
}

/// Model for Posts!
model Post {
  id        Int         @id @default(autoincrement())
  bigIntId  BigInt      @unique
  createdAt DateTime    @unique @default(now())
  updatedAt DateTime    @updatedAt
  /// Title for the post
  title     String
  /// Content for the post
  content   String?
  /// Has this post been published?
  published Boolean     @default(false)
  author    User        @relation(fields: [authorId], references: [id])
  comments  Comment[]
  authorId  Int
  media     PostMedia[]
  views     Decimal     @default(0)
}

model Media {
  id           Int         @id @default(autoincrement())
  url          String
  posts        PostMedia[]
  uploadedBy   User?       @relation(fields: [uploadedById], references: [id])
  uploadedById Int?
}

model PostMedia {
  id      Int   @id @default(autoincrement())
  post    Post  @relation(fields: [postId], references: [id])
  media   Media @relation(fields: [mediaId], references: [id])
  postId  Int
  mediaId Int
  order   Int

  @@unique([postId, mediaId])
}

model Comment {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @unique @default(now())
  content   String
  author    User     @relation(fields: [authorId], references: [id])
  post      Post     @relation(fields: [postId], references: [id])
  authorId  Int
  postId    Int
}

model Profile {
  id     Int     @id @default(autoincrement())
  bio    String?
  user   User    @relation(fields: [userId], references: [id])
  userId Int     @unique
}

model User {
  id        Int       @id @default(autoincrement())
  email     String    @unique
  name      String?
  posts     Post[]
  comments  Comment[]
  profile   Profile?
  followers Follow[]  @relation("followers")
  following Follow[]  @relation("following")
  Media     Media[]
}

model Follow {
  fromId Int
  toId   Int
  from   User @relation("following", fields: [fromId], references: [id])
  to     User @relation("followers", fields: [toId], references: [id])

  @@id([fromId, toId], name: "compositeID")
}

model Unrelated {
  id   Int     @id @default(autoincrement())
  name String?
}

model WithID {
  id                  String                @id
  FindUniqueRelations FindUniqueRelations[]
}

model WithUnique {
  id                  String                @unique
  FindUniqueRelations FindUniqueRelations[]
}

model WithCompositeID {
  a String
  b String

  FindUniqueRelations FindUniqueRelations[]

  @@id([a, b])
}

model WithCompositeUnique {
  a String
  c String? @unique
  b String

  FindUniqueRelations FindUniqueRelations[]

  @@unique([a, c])
  @@unique([a, b])
}

model FindUniqueRelations {
  id String @id

  withID_id String
  withID    WithID @relation(fields: [withID_id], references: [id])

  withUnique_id String
  withUnique    WithUnique @relation(fields: [withUnique_id], references: [id])

  withCompositeID_a String
  withCompositeID_b String
  withCompositeID   WithCompositeID @relation(fields: [withCompositeID_a, withCompositeID_b], references: [a, b])

  withCompositeUnique_a String
  withCompositeUnique_b String
  withCompositeUnique   WithCompositeUnique @relation(fields: [withCompositeUnique_a, withCompositeUnique_b], references: [a, b])
}
