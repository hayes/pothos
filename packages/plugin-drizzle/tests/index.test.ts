import { printSchema } from 'graphql';
import { schema } from './example/schema';
import { schema as postgresSchema } from './postgres/schema';

describe('drizzle plugin', () => {
  it('produces the expected schema', () => {
    expect(printSchema(schema)).toMatchSnapshot();
  });

  it('produces the expected postgres schema', () => {
    expect(printSchema(postgresSchema)).toMatchSnapshot();
  });
});
