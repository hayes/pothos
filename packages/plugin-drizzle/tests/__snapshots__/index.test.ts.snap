// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`drizzle plugin > produces the expected schema 1`] = `
""""
A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the \`date-time\` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
"""
scalar DateTime

type Error {
  message: String
  stack: String
}

type Mutation {
  commentOnPost(input: MutationCommentOnPostInput!): comments
  createPost(input: MutationCreatePostInput!): Post
  likePost(postId: ID!): Post
  publishPost(postId: ID!): Post
}

input MutationCommentOnPostInput {
  postId: ID!
  text: String!
}

input MutationCreatePostInput {
  category: String!
  content: String!
  published: Boolean
  title: String!
}

interface Node {
  id: ID!
}

type PageInfo {
  endCursor: String
  hasNextPage: Boolean!
  hasPreviousPage: Boolean!
  startCursor: String
}

type Post {
  author: User
  category: String
  comments: [comments!]
  content: String
  createdAt: DateTime
  id: ID
  likes: Int
  published: Boolean
  title: String
  updatedAt: DateTime
}

type Query {
  me: Viewer
  node(id: ID!): Node
  nodes(ids: [ID!]!): [Node]!
  post(id: ID!): Post
  postWithErrors(id: ID!): QueryPostWithErrorsResult
  posts(after: String, before: String, category: String, first: Int, last: Int): QueryPostsConnection
  user(id: ID!): User
  userWithInput(input: QueryUserWithInputInput!): User
  users(id: ID!): [User!]
}

union QueryPostWithErrorsResult = Error | QueryPostWithErrorsSuccess

type QueryPostWithErrorsSuccess {
  data: Post!
}

type QueryPostsConnection {
  edges: [QueryPostsConnectionEdge]
  pageInfo: PageInfo!
}

type QueryPostsConnectionEdge {
  cursor: String!
  node: Post
}

input QueryUserWithInputInput {
  id: ID!
}

type User implements Node {
  bio: String
  email: String
  firstName: String
  fullName: String
  id: ID!
  isAdmin: Boolean!
  lastName: String
  posts(limit: Int, offset: Int): [Post!]
  postsConnection(after: String, before: String, first: Int, last: Int): UserPostsConnection
  unOrderedPostsConnection(after: String, before: String, first: Int, last: Int): UserUnOrderedPostsConnection
  viewer: Viewer
}

type UserPostsConnection {
  edges: [UserPostsConnectionEdge]
  pageInfo: PageInfo!
}

type UserPostsConnectionEdge {
  cursor: String!
  node: Post
}

type UserUnOrderedPostsConnection {
  edges: [UserUnOrderedPostsConnectionEdge]
  pageInfo: PageInfo!
}

type UserUnOrderedPostsConnectionEdge {
  cursor: String!
  node: Post
}

type Viewer {
  comments(after: String, before: String, first: Int, last: Int): ViewerCommentsConnection
  drafts: [Post!]
  id: ID
  roles: [String!]
  user: User!
  username: String
}

type ViewerCommentsConnection {
  edges: [ViewerCommentsConnectionEdge]
  pageInfo: PageInfo!
}

type ViewerCommentsConnectionEdge {
  cursor: String!
  node: comments
}

type comments {
  author: User
  id: ID
  post: Post
  text: String
}"
`;
